internal/quaiapi/quai_api.go

```diff
@@ -634,3 +635,17 @@ func (s *PublicBlockChainQuaiAPI) GetSubManifest(ctx context.Context, raw json.R
 	}
 	return manifest, nil
 }
+
+type SendPendingEtxsToDomArgs struct {
+	Header types.Header
+	Etxs   []types.Transactions // Collection of ETXs from each subordinate context
+}
+
+func (s *PublicBlockChainQuaiAPI) SendPendingEtxsToDom(ctx context.Context, raw json.RawMessage) error {
+	var args SendPendingEtxsToDomArgs
+	if err := json.Unmarshal(raw, &args); err != nil {
+		return err
+	}
+	log.Info("Received subordinate ETXs", "BlockHash: ", args.Header.Hash())
+	return s.b.AddPendingEtxs(&args.Header, args.Etxs)
```

Comment:
No need for the `&args.Header` here. Unmarshal uses the memory address of args to decode to. You can use `args.Header` and `args.Etxs`.

Reply:
Also since the array of transactions is a pointer, we won't get the deep copy. If this method is being used to send cross contexts. Might need a for loop in that case. Just a note.

Reply:
> No need for the &args.Header here

Not sure what you mean. `AddPendingEtxs()` takes header by reference `*types.Header`. So I need to pass a pointer in. Trying to pass by value results in an error.

> Also since the array of transactions is a pointer, we won't get the deep copy. If this method is being used to send cross contexts. Might need a for loop in that case. Just a note.

Yes, this intentional. No need for deep copy here. This method does not go across contexts.